src/main/java/com/amazon/opendistroforelasticsearch/sql/optimizer/OptimizeRule.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.optimizer;
+
+import java.sql.SQLFeatureNotSupportedException;
+
+/**
+ * Query Optimize Rule
```

Comment:
This documentation is not very useful, either give more info what it does, or delete completely if the interface is pretty straightforward.